File: app/src/main/java/com/marlon/portalusuario/database/notifications/PunDAO.java
package com.marlon.portalusuario.database.notifications;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.marlon.portalusuario.PUNotifications.PUNotification;

import java.util.List;

@Dao
public interface PunDAO {

    @Insert
    void insertPUNotification(PUNotification post);

    @Update
    void updatePUNotification(PUNotification post);

    @Delete
    void deletePUNotification(PUNotification post);

    @Query("SELECT * FROM notifications")
    LiveData<List<PUNotification>> getAllPUNotification();

    @Query("DELETE FROM notifications")
    void deleteAllPUNotification();
}

File: app/src/main/java/com/marlon/portalusuario/database/notifications/PunDAO.kt
package com.marlon.portalusuario.database.notifications

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.marlon.portalusuario.PUNotifications.PUNotification

@Dao
interface PunDAO {
    @Insert
    fun insertPUNotification(post: PUNotification)

    @Update
    fun updatePUNotification(post: PUNotification)

    @Delete
    fun deletePUNotification(post: PUNotification)

    @Query("SELECT * FROM notifications")
    fun getAllPUNotification(): LiveData<List<PUNotification>>

    @Query("DELETE FROM notifications")
    fun deleteAllPUNotification()
}
